spec.emu

```diff
@@ -211,33 +217,30 @@ contributors: Ujjwal Sharma, Younies Mahmoud
-        1. Set _opt_.[[localeMatcher]] to _matcher_.
         1. Let _numberingSystem_ be ? GetOption(_options_, `"numberingSystem"`, `"string"`, `undefined`, `undefined`).
         1. If _numberingSystem_ does not match the Unicode Locale Identifier `type` nonterminal, throw a `RangeError` exception.
-        1. Set _opt_.[[nu]] to _numberingSystem_.
-        1. Let _localeData_ be %DurationFormat%.[[LocaleData]].
-        1. Let _r_ be ResolveLocale(%DurationFormat%.[[AvailableLocales]], _requestedLocales_, _opt_, %DurationFormat%.[[RelevantExtensionKeys]], _localeData_).
+        1. Let _opt_ be the Record { [[localeMatcher]]: _matcher_, [[nu]]: _numberingSystem_ }.
+        1. Let _r_ be ResolveLocale(%DurationFormat%.[[AvailableLocales]], _requestedLocales_, _opt_, %DurationFormat%.[[RelevantExtensionKeys]], %DurationFormat%.[[LocaleData]]).
         1. Let _locale_ be r.[[locale]].
         1. Set _durationFormat_.[[Locale]] to _locale_.
         1. Set _durationFormat_.[[NumberingSystem]] to _r_.[[nu]].
         1. Let _style_ be ? GetOption(_options_, `"style"`, `"string"`, &laquo; `"long"`, `"short"`, `"narrow"`, `"digital"` &raquo;, `"long"`).
         1. Set _durationFormat_.[[Style]] to _style_.
         1. Set _durationFormat_.[[DataLocale]] to _r_.[[dataLocale]].
-        1. Let _yearsOptions_ be ? GetUnitOptions(`"years"`, _options_, _style_).
+        1. Let _yearsOptions_ be ? GetUnitOptions(`"years"`, _options_, _style_, _style_).
         1. Set _durationFormat_.[[YearsStyle]] to _yearsOptions_.[[Style]].
         1. Set _durationFormat_.[[YearsDisplay]] to _yearsOptions_.[[Display]].
-        1. Let _monthsOptions_ be ? GetUnitOptions(`"months"`, _options_, _style_).
+        1. Let _monthsOptions_ be ? GetUnitOptions(`"months"`, _options_, _style_, _durationFormat_.[[YearsStyle]]).
         1. Set _durationFormat_.[[MonthsStyle]] to _monthsOptions_.[[Style]].
         1. Set _durationFormat_.[[MonthsDisplay]] to _monthsOptions_.[[Display]].
-        1. Let _weeksOptions_ be ? GetUnitOptions(`"weeks"`, _options_, _style_).
+        1. Let _weeksOptions_ be ? GetUnitOptions(`"weeks"`, _options_, _style_, _durationFormat_.[[MonthsStyle]]).
         1. Set _durationFormat_.[[WeeksStyle]] to _weeksOptions_.[[Style]].
         1. Set _durationFormat_.[[WeeksDisplay]] to _weeksOptions_.[[Display]].
-        1. Let _daysOptions_ be ? GetUnitOptions(`"days"`, _options_, _style_).
+        1. Let _daysOptions_ be ? GetUnitOptions(`"days"`, _options_, _style_, _durationFormat_.[[WeeksStyle]]).
         1. Set _durationFormat_.[[DaysStyle]] to _daysOptions_.[[Style]].
         1. Set _durationFormat_.[[DaysDisplay]] to _daysOptions_.[[Display]].
-        1. Let _hoursOptions_ be ? GetUnitOptions(`"hours"`, _options_, _style_).
+        1. Let _hoursOptions_ be ? GetUnitOptions(`"hours"`, _options_, _style_, _durationFormat_.[[DaysStyle]]).
```

Comment:
I wasn't sure what to use in these calls to `GetUnitOptions`, but it has 4 mandatory parameters and only 3 arguments were passed, so something must go here.